upgrade.py

```diff
@@ -487,6 +487,34 @@ def buildRestoreList(self, src_base):
         return restore_list
 
     completeUpgradeArgs = ['mounts', 'installation-to-overwrite', 'primary-disk', 'backup-partnum', 'logs-partnum', 'net-admin-interface', 'net-admin-bridge', 'net-admin-configuration']
+
+    def convert_interface_rename_data(self, src_file_path, dest_file_path):
+        """
+        Convert data of interface-rename to the data consumed by networkd.
+        """
+        net_devs = netutil.net_devs_of_last_boot(src_file_path)
+        if net_devs is None:
+            logger.error('Failed to parse the interface-rename dynamic rules.')
+            return False
+        else:
+            pattern = r'eth(\d+)'
+            lines = []
+            for (name, mac) in net_devs:
+                match = re.search(pattern, name)
+                if match:
+                    position = match.group(1)
+                    line = f"{position}:mac=\"{mac}\"\n"
+                    logger.log(f"Converted interface-rename data: {line}")
+                    lines += line
+                else:
+                    logger.error(f'Network device name {name} is not like ethN.')
+                    return False
+
+            with open(dest_file_path, mode='w', encoding='utf-8') as f:
+                f.writelines(lines)
```

Comment:
Instead of building up a list of lines and then writing them like this, maybe it would be better to just write the file line by line?